### PR TITLE
Handle 207 better

### PR DESCRIPTION
**Pull request by smusali**
When `logdna-agent` sends a batch of lines containing invalid ones, `logdna-agent` can't understand `207` and in order to provide visibility over the invalid lines, this `Pull Request` is adding debug log statements to specify which log lines are invalid.

This `Pull Request` also renames some variables related to requesting and/or handling responses to make more readable.

Considering `logdna-agent` doesn't use any transport other than `http`, there is no need for `config.TRANSPORT` variable and checks.

### CODE REVIEW

File: lib/linebuffer.js [hunk 1]
```diff
@@ -45,121 +45,123 @@ const needsRetry = (error, response, body) => {
         httpErrorTime = null;
     }
 
+    buffer = [];
```
Comment (weizou19) [inline lib/linebuffer.js]: Why this line is needed?

Comment (smusali) [reply lib/linebuffer.js]: to empty after each flush

Comment (dchai76) [reply lib/linebuffer.js]: I think Wei is asking: doesn't line 70 do that already?

Comment (smusali) [reply lib/linebuffer.js]: yup - discussed already - forgot to update: remove that part

**Comment by smusali** [comment]
Updated, @dchai76 @weizou19!

**Comment by dchai76** [comment]
In Phabricator land we ask for a Test Plan - how did you test this to show that the changes work. Let me ask for that here also. What did you do to test?

**Comment by smusali** [comment]
> In Phabricator land we ask for a Test Plan - how did you test this to show that the changes work. Let me ask for that here also. What did you do to test?

So, @dchai76, well, this is very interesting question and there is no direct answer from me. Basically, the integration testing with test account is going to test this case.

Also, this is small addition to the code which is not affecting the current logflow and this small block was tested with the sample response body.

The other thing is `207` response code is temporary and going to be gone soon.

**Pull request merged**